xds/src/main/java/com/linecorp/centraldogma/xds/internal/XdsResourceWatchingService.java

```diff
@@ -84,26 +89,36 @@ protected void init() {
             watchingGroups.add(groupName);
             final Revision normalizedRevision = repository.normalizeNow(Revision.HEAD);
             logger.info("Creating xDS resources from {} at revision: {}", groupName, normalizedRevision);
-            final Map<String, Entry<?>> entries = repository.find(normalizedRevision, pathPattern()).join();
-            for (Entry<?> entry : entries.values()) {
-                if (entry.type() != EntryType.JSON || !entry.hasContent()) {
-                    continue;
+            final CompletableFuture<Map<String, Entry<?>>> findFuture =
+                    repository.find(normalizedRevision, pathPattern());
+            futures.add(findFuture.handleAsync((entries, cause) -> {
+                // Executed by the repository worker thread.
+                if (cause != null) {
+                    logger.warn("Unexpected exception while finding {} at revision: {}",
+                                groupName, normalizedRevision, cause);
+                    return null;
                 }
-                final String path = entry.path();
-                final String contentAsText = entry.contentAsText();
-                try {
-                    handleXdsResource(path, contentAsText, groupName);
-                } catch (Throwable t) {
-                    throw new RuntimeException("Unexpected exception while building an xDS resource from " +
-                                               groupName + path, t);
+                for (Entry<?> entry : entries.values()) {
+                    if (entry.type() != EntryType.JSON || !entry.hasContent()) {
+                        continue;
+                    }
+                    final String path = entry.path();
+                    final String contentAsText = entry.contentAsText();
+                    try {
+                        handleXdsResource(path, contentAsText, groupName);
+                    } catch (Throwable t) {
+                        logger.warn("Unexpected exception while building an xDS resource from {}.",
```

Comment:
ditto

Question) It seems like this behavior has changed from failing the startup to just logging a warning.
When restarting, are we expected to check for this log? or would just explicitly throwing a `RuntimeException` make more sense?

Reply:
Let me just leave a warning log in this case.
This could happen when an admin made a commit via `ContentServiceV1` and I think it doesn't make sense to stop Central Dogma because of just one file. We can just ignore it. 😉